### ISSUE TITLE
Problem in implemeting the ITRF93 to J2000 rotation in TUDAT estimation algorithm

### ISSUE THREAD

**Issue by Pugazhenthi-Sivasankar** [Tudat/tudat#426]
Hello everyone,
In the 'earthOrbiterBasicStateEstimation.cpp', I changed the rotation model of Earth from the default setting (IAU_Earth to ECLIPJ2000) to my case that required more accuracy (ITRF93 to J2000). When I executed the script, I got the following error:

[itrf93_error.txt](https://github.com/Tudat/tudat/files/2377085/itrf93_error.txt)

I traced back and found that the string value 'fixedReferenceFrame' in sphericalHarmonicGravityField.h was IAU_Earth instead of ITRF93. I tried changing it the following way:

accelerationsOfVehicle[ "Earth" ].push_back( boost::make_shared< SphericalHarmonicAccelerationSettings >( 50, 50 ), "ITRF93" );

This resulted in the following compilation errors:

[itrf93_comp_err.txt](https://github.com/Tudat/tudat/files/2377095/itrf93_comp_err.txt)
The relevant script for the above situation is given in the link below:
https://github.com/Pugazhenthi-Sivasankar/Pugazh1

In another script, I tried with only the central gravity model to check the changes in the earth rotation model. I tried to print the rotation matrix and its derivative between the frames ITRF93 and J2000 as output. But I got the following error:

[rot_test_err.txt](https://github.com/Tudat/tudat/files/2377109/rot_test_err.txt)

The 'tisbod.c' file says that a pck binary file or text file corresponding to ITRF93 is missing. However, I have added the 'earth_latest_high_prec.bpc' and the function "spice_interface::loadStandardSpiceKernels( );" loads other necessary files. The relevant script for the above situation is given in the link below:
https://github.com/Pugazhenthi-Sivasankar/Pugazh1

Help would be appreciated!!

**Comment by DominicDirkx** [comment]
Hi Pugazh, 

The problem seems to be that the associatedReferenceFrame_ member of the SphericalHarmonicsGravityFieldSettings object for Earth is set to IAU_Earth, not ITRF93,

Dominic

**Comment by Pugazhenthi-Sivasankar** [comment]
I found the member functions 'getAssociatedReferenceFrame()' and 'resetAssociatedReferenceFrame()' in the file 'createGravity.h'. I attempted to obtain the existing reference frame before attempting to change it to ITRF93 and it resulted in the following compilation errors:

[asso_ref_frame_comp_err.txt](https://github.com/Tudat/tudat/files/2381344/asso_ref_frame_comp_err.txt)

I tried to access the member function in other ways, but they all resulted in similar compilation errors. The relevant algorithm is provided in the link below:
https://github.com/Pugazhenthi-Sivasankar/Pugazh1

**Comment by Pugazhenthi-Sivasankar** [comment]
My apologies and thanks to your suggestion I figured out how to get and reset the associatedReferenceFrame for the spherical harmonics gravity field settings. I checked the associatedReferenceFrame before and after resetting it to ITRF93. The output shows a change from IAU_Earth to ITRF93. But, the same run-time error appears again. I have attached the error message and the script for your perusal:
[itrf93_err2.txt](https://github.com/Tudat/tudat/files/2383301/itrf93_err2.txt)
https://github.com/Pugazhenthi-Sivasankar/Pugazh1
